src/pages/Dashboard.tsx

```diff
@@ -77,28 +101,22 @@ class Dashboard extends Component<Props, DashboardState> {
       </Card>
     );
   }
+
   public render() {
     const { location, t } = this.props;
     const applicationConfig = get(this.state, "applicationConfig");
     const showModal = get(this.state, "showModal");
     const lastMigrationsRefresh = get(this.state, "lastMigrationsRefresh");
+    const scheduledInfoBanners = get(
+      this.state,
+      "scheduledInfoBanners",
+      [] as readonly ScheduledInfo[]
+    );
     const toasts = get(this.state, "toasts", [] as readonly ToastrItem[]);
+
     return (
       <>
-        <div className="p-3">
-          {/* APIM V2 MIGRATION - TEMPORARY DISCLAIMER */}
-          <Alert color="warning">
-            Ti informiamo che dalle 08:00 alle 24:00 di lunedì 31 luglio verrà
-            effettuato un importante aggiornamento del sistema di gestione delle
-            API. Durante questo periodo sarà comunque{" "}
-            <strong>possibile inviare messaggi</strong>, ma saranno
-            temporaneamente non disponibili le API, e i servizi sul Developer
-            Portal e sull'Area Riservata.
-            <br />
-            Ci scusiamo per il disagio e ti ringraziamo per la collaborazione.
-          </Alert>
-          {/* APIM V2 MIGRATION - TEMPORARY DISCLAIMER */}
-        </div>
+        <ScheduledInfoViewer items={scheduledInfoBanners} />
```

Comment:
You are passing the full list to a component named ScheduledInfoViewer with inside the logic to display the "active" banner.
Why you prefer this over having the ScheduledInfoViewer as a pure presentational component and move the "check the active info" logic here?

Reply:
Because in my intentions **"ScheduledInfoViewer"** implies a component that displays _(Viewer)_ an information _(Info)_ based on a schedule _(Scheduled)_.
For this reason it comes natural to me to think that this component has the logic of understanding what to show based on the schedules and based on the current date of visualization on the client side (browser).

Differently, if I had thought of having this logic outside the component, I would have created a "lighter" component _(maybe just <InfoViewer/>)_ with the disadvantage of having to absorb the scheduling selection logic in the page _(and in general in all pages)_ which must use this component.